terraform/modules/dns/main.tf

```
@@ -0,0 +1,17 @@
+resource "dns_zone" "this" {
+  domain = var.zone_domain
+}
+
+module "record_sets" {
+  source = "./modules/record_sets"
+  for_each = {
+    for record in var.recordsets : join("/", [record.name, record.type]) => record
+  }
+
+  from_parent = true
+  domain = dns_zone.this.domain
```

```suggestion
  zone_domain = dns_zone.this.domain
```
L'attribut 'from_parent' n'existe pas dans  la définition du module